tdns/sanitize_data.go

```
@@ -140,13 +183,17 @@ func deepCopyAndSanitize(v interface{}) interface{} {
 				// If the sanitized value is nil, set to zero value
 				copyField.Set(reflect.Zero(field.Type()))
 			} else {
+				// log.Printf("deepCopyAndSanitize: struct: field: %v (type %T)", field, field.Type().String())
 				// Use a try-catch approach with defer/recover
 				func() {
+					// log.Printf("deepCopyAndSanitize: struct: field: %s", val.Type().Field(i).Name)
 					defer func() {
 						if r := recover(); r != nil {
-							log.Printf("Warning: Failed to set field %s: %v", val.Type().Field(i).Name, r)
+							log.Printf("*** Warning: struct: field type: %s", reflect.TypeOf(v).String())
+							log.Printf("*** Warning: Failed to set field #%s %s: %v", val.Type().Field(i).Name, i, r)
```

golangci-lint (1.64.8)

printf: log.Printf format %s has arg i of wrong type int

(govet)